dm/templates/unmanaged_instance_group/unmanaged_instance_group.py

```diff
@@ -0,0 +1,128 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" This template creates an unmanaged instance group. """
+
+def set_optional_property(receiver, source, property_name):
+    """ If set, copies the given property value from one object to another. """
+
+    if property_name in source:
+        receiver[property_name] = source[property_name]
+
+
+def generate_instance_url(project, zone, instance):
+    """ Format the resource name as a resource URI. """
+
+    is_self_link = '/' in instance or '.' in instance
+
+    if is_self_link:
+        instance_url = instance
+    else:
+        instance_url = 'projects/{}/zones/{}/instances/{}'
+        instance_url = instance_url.format(project, zone, instance)
+
+    return instance_url
+
+
+def generate_config(context):
+    """ Entry point for the deployment resources. """
+
+    properties = context.properties
+    name = properties.get('name', context.env['name'])
+    project_id = properties.get('project', context.env['project'])
+    zone = properties.get('zone')
+
+    # Network formatting
+    if 'network' in properties:
+        network_name = properties.get('network')
+        if not '/' in network_name:
+            network_propertie = { 'network': 'global/networks/{}'.format(network_name) }
+        else:
+            network_propertie = { 'network': network_propertie }
+    else:
+        network_propertie = {}
+
+    properties.update(network_propertie)
+
+    # Create unmanaged instance group resource
+    umig_properties = {
+        'name': name,
+        'project': project_id,
+        'zone': zone
+    }
+
+    known_properties = [
+        'description',
+        'namedPorts',
+        'region',
+        'network',
+    ]
+
+    for prop in known_properties:
+        set_optional_property(umig_properties, properties, prop)
+        
+    umig_resources = [
+        {
+            'name': name,
+            'type': 'gcp-types/compute-v1:instanceGroups',
+            'properties': umig_properties
+        },
+    ]
+
+    # If instances are specified, add them to group.
+    # https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroups/addInstances
+    instance_provider_type = 'gcp-types/compute-v1:compute.instanceGroups.addInstances'
+    instances = properties.get('instances', [])
+    if instances:
+        formatted_instances = []
+        for instance_name in instances:
+            instance_url = generate_instance_url(
+                project_id,
+                zone,
+                instance_name.get('instance')
+            )
+            instance = { 'instance': instance_url }
+            formatted_instances.append(instance)
+
+        add_instances = {
+            'name': '{}-addinstances'.format(name),
+            'action': (instance_provider_type),
```

Comment:
This will try to add every instance at every update. This needs to have a runtime policy for create only

Reply:
@ocsig 
I can add runtime policy and change input structure like that:
```
instances:
  add: 
  - instance-1
  - instance-2
  remove:
  - instance-3
```
Is that acceptable?